dlint/linters/input_password_ask.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+
+from . import base
+
+
+class InputPasswordUseLinter(base.BaseLinter):
+    """This linter looks for use of the Python "input" function with a string
+    argument containing the word "password" in any format.
+    """
+    off_by_default = False
+
+    _code = 'DUO139'
+    _error_tmpl = 'DUO139 avoid using "input" to ask for password'
+
+    def visit_Call(self, node):
+
+        if (node.func.id == "input"):
+            value = self._get_arg_or_kwarg(node)
+
+            if "password" in value.lower():
+                self.results.append(
+                    base.Flake8Result(
+                        lineno=node.lineno,
+                        col_offset=node.col_offset,
+                        message=self._error_tmpl
+                    )
+                )
+
+    def _get_arg_or_kwarg(self, node):
+        if node.args:
+            return node.args[0].value
```

Comment:
You should also check that `node.args` and `node.keywords` are not empty before checking `[0]`. This would produce a runtime error in the code because `input` takes 1 argument, but you never know what code you'll find in the wild. In other words, static analysis will often see code that hasn't been executed, or has all kinds of weird bugs, so it's best to be defensive :)